[JSC] Use Butterfly::clearOptimalVectorLengthGap for JSArray::tryCreate's butterfly initialization

Constellation opened the pull request:
#### b1feea69abb78ccf280cfad92af10438c147ee6f
<pre>
[JSC] Use Butterfly::clearOptimalVectorLengthGap for JSArray::tryCreate&apos;s butterfly initialization
<a href="https://bugs.webkit.org/show_bug.cgi?id=296842">https://bugs.webkit.org/show_bug.cgi?id=296842</a>
<a href="https://rdar.apple.com/157353749">rdar://157353749</a>

Reviewed by Sosuke Suzuki.

Use Butterfly::clearOptimalVectorLengthGap as butterfly is not set to
JSArray yet.

* Source/JavaScriptCore/runtime/JSArray.h:
(JSC::JSArray::tryCreate):

Canonical link: <a href="https://commits.webkit.org/298174@main">https://commits.webkit.org/298174@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3233905b603ee899657f310a9cf710f67dfd1bd3

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/114490 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/34236 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/24699 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/120656 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/65209 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/d82686fa-f384-4389-81bb-b3b6f6301cb4) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/34865 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/42796 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/87014 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/41919 "") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/b86c698d-f21b-456f-9217-a0abf39db444) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/117438 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/27777 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/102819 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/67405 "Passed tests") | | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/26961 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/20946 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/64336 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/106886 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/97154 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/21060 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/123861 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/113053 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/41503 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/30971 "Found 1 new test failure: fast/block/transparent-outline-with-and-without-border-radius.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/95831 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/41880 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/99010 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/95615 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/40779 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/18616 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/37536 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18342 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/41382 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/46892 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/137270 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/40970 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/36706 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/44277 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/42720 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

sosukesuzuki reviewed (changes requested):
Build error on EWS for watch and playstation
```
/Volumes/Data/worker/watchOS-11-Build-EWS/build/Source/JavaScriptCore/runtime/Butterfly.h:244:31: error: inline function 'JSC::Butterfly::clearOptimalVectorLengthGap' is not defined [-Werror,-Wundefined-inline]
  244 |     ALWAYS_INLINE static void clearOptimalVectorLengthGap(IndexingType, Butterfly*, unsigned optimalVectorLength, unsigned vectorLength);
      |                               ^
In file included from /Volumes/Data/worker/watchOS-11-Build-EWS/build/WebKitBuild/Release-watchos/DerivedSources/JavaScriptCore/unified-sources/UnifiedSource1.cpp:1:
In file included from /Volumes/Data/worker/watchOS-11-Build-EWS/build/WebKitBuild/Release-watchos/DerivedSources/JavaScriptCore/BytecodeDumperGenerated.cpp:31:
In file included from /Volumes/Data/worker/watchOS-11-Build-EWS/build/WebKitBuild/Release-watchos/DerivedSources/JavaScriptCore/BytecodeStructs.h:31:
In file included from /Volumes/Data/worker/watchOS-11-Build-EWS/build/Source/JavaScriptCore/bytecode/ArrayAllocationProfile.h:29:
/Volumes/Data/worker/watchOS-11-Build-EWS/build/Source/JavaScriptCore/runtime/JSArray.h:254:20: note: used here
  254 |         Butterfly::clearOptimalVectorLengthGap(indexingType, butterfly, vectorLength, 0);
      |                    ^
1 error generated.
```

Constellation committed:
[JSC] Use Butterfly::clearOptimalVectorLengthGap for JSArray::tryCreate's butterfly initialization
https://bugs.webkit.org/show_bug.cgi?id=296842
rdar://157353749

Reviewed by Sosuke Suzuki.

Use Butterfly::clearOptimalVectorLengthGap as butterfly is not set to
JSArray yet.

* Source/JavaScriptCore/runtime/JSArray.h:
(JSC::JSArray::tryCreate):

Canonical link: https://commits.webkit.org/298174@main
